### PR TITLE
Implement Junction::is_intersection API 

### LINKED ISSUE #510: Implement is_intersection API
New API is_intersection should be implemented.

To get this info, a new road_geometry_configuration param should be set to enable userData XODR tags for Junctions.

If the tag is not set, no info can be inferred to ensure there is an intersection, otherwise, we want to follow this criteria:

* STEP1: Junction `userData` override
  * Inspect junction-level userData elements with `code="junctionType"`:
    * `value="intersection"` => junction is treated as an intersection.
    * `value="nonIntersection"` => junction is treated as not an intersection.
    * Other / unexpected values => fall through to STEP2.
* STEP2: Road type check:
  * Inspect the OpenDRIVE road type at the relevant location:
    * If road type is `"motorway"` => not an intersection.
      * Rationale: motorway on-ramps/off-ramps are treated as merge/diverge branches, not intersections.
    * If road type is not `"motorway"` => fall through to STEP3.
* STEP3: Connection count fallback:
  * If OpenDRIVE connection size is > 4, flag this automatically as an intersection.
  * Otherwise, not an intersection.

### PR DESCRIPTION
**Pull request by santiago-lopez-ekumen**
# 🎉 New feature

Closes #510 

## Summary
Implement `Junction::is_intersection()` API based on userData XODR node info in XODR Junction nodes.

### Detailed changelog
  - Added configurable intersection detection in road geometry config (`use_userdata_intersections`), and propagated it through parser and builder.
   - Extended XODR junction parsing to keep all `<userData>` siblings (not just the first).
   - Updated junction intersection override logic to scan all stored userData entries for `code="junctionType"`
     - uses `value=intersection` / `value=nonIntersection` when only one is provided,
     - treats conflicting values (multiple `<userData>` with different `code="junctionType"` values) as `std::nullopt` and logs a warning message,
     - then falls back to existing motorway/connection-count rules as described in #510 .
   - Added tests for: 
     - `road_geometry_configuration` parsing,
     - parser support for multiple junction `userData` entries,
     - builder override precedence and conflict handling.

### Suggested review order
   1. builder/road_geometry_configuration.h (and .cc) + builder/road_network_builder.cc + builder/xodr_parser_configuration.cc (feature flag)
   2. xodr/junction.h (and .cc) + xodr/parser.cc (userData parsing)
   3. builder/builder_tools.h (and .cc) (intersection decision logic)
   4. test/regression/xodr/parser_test.cc (parser behavior)
   5. test/regression/builder/builder_tools_test.cc and config tests (logic + edge cases)

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [x] Updated documentation (as needed)

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/maliput_malidrive/builder/builder_tools.cc [hunk 1]
```diff
@@ -237,6 +238,99 @@ std::vector<maliput::api::LaneEnd> SolveLaneEndsWithinJunction(
   return SolveLaneEndsForConnectingRoad(rg, xodr_lane_properties, road_headers, connection_type);
 }
 
+std::optional<bool> DetermineJunctionIntersectionFromXodr(
+    const xodr::Junction& junction, const std::map<xodr::RoadHeader::Id, xodr::RoadHeader>& road_headers) {
+  struct IntersectionOverrideResult {
+    std::optional<bool> value{std::nullopt};
+    bool has_conflict{false};
+  };
+  // Returns true if the junction is an intersection, false if it is not, and nullopt if it cannot be determined.
```
Comment (jakeaboganda) [inline src/maliput_malidrive/builder/builder_tools.cc]: nit: this comment should probably be moved up, as it is a comment for the DetermineJunctionIntersectionFromXodr itself

File: src/maliput_malidrive/builder/builder_tools.cc [hunk 2]
```diff
@@ -237,6 +238,99 @@ std::vector<maliput::api::LaneEnd> SolveLaneEndsWithinJunction(
...
+      while (!nodes.empty()) {
+        tinyxml2::XMLElement* node = nodes.back();
+        nodes.pop_back();
+        const char* code = node->Attribute("code");
+        if (code != nullptr && std::string(code) == "junctionType") {
+          const char* value = node->Attribute("value");
+          if (value != nullptr) {
+            const std::string junction_type_value(value);
+            if (junction_type_value == "intersection") {
+              has_intersection = true;
```
Comment (jakeaboganda) [inline src/maliput_malidrive/builder/builder_tools.cc:280]: We could probably exit early on when there are conflicts

File: src/maliput_malidrive/builder/builder_tools.h [hunk 3]
```diff
@@ -142,6 +142,20 @@ std::vector<maliput::api::LaneEnd> SolveLaneEndsWithinJunction(
...
+/// OpenDRIVE-derived data and first-match-wins priority:
+/// 1. junction userData override (`code="junctionType"` with value
+///    `intersection` / `nonIntersection`).
+/// 2. motorway road type participation (if any participating road is motorway,
+///    classify as non-intersection).
+/// 3. connection-count fallback (`connections.size() > 4`).
+///
+/// @param junction Junction data from the XODR map.
+/// @param road_headers Road headers from the XODR map.
+/// @returns Intersection classification according to the priority rules.
```
Comment (jakeaboganda) [inline src/maliput_malidrive/builder/builder_tools.h]: Needs a comment on what a null return means.

File: src/maliput_malidrive/xodr/parser_configuration.h [hunk 4]
```diff
@@ -55,6 +55,9 @@ struct ParserConfiguration {
 
   /// When active, the parser will support userData XODR nodes.
   bool use_userdata_traffic_direction{false};
+
+  /// When active, the parser will support intersection-related userData XODR nodes.
```
Comment (jakeaboganda) [inline src/maliput_malidrive/xodr/parser_configuration.h:59]: nit: perhaps needs a reference to DetermineJunctionIntersectionFromXodr()

File: test/regression/builder/builder_tools_test.cc [hunk 5]
```diff
@@ -916,6 +917,113 @@ TEST_F(ResolveLaneIdsOmitNonDrivableTest, ExplicitValidityFilteredByBuiltLanes)
   EXPECT_NE(std::find(result.begin(), result.end(), maliput::api::LaneId("1_0_-1")), result.end());
 }
```
Comment (jakeaboganda) [inline test/regression/builder/builder_tools_test.cc:919]: needs a conflict test?

**Review by jakeaboganda — COMMENTED** [review]
Thank you so much! Added some small nits on the PR. Overall looks good. Please don't treat this as a blocking review.

File: src/maliput_malidrive/builder/builder_tools.cc [hunk 6]
```diff
@@ -237,6 +238,99 @@ std::vector<maliput::api::LaneEnd> SolveLaneEndsWithinJunction(
...
+    return {};
+  };
+
+  auto has_any_motorway_participating_road = [&road_headers](const xodr::Junction& xodr_junction) -> bool {
+    std::set<std::string> participating_road_ids{};
+    for (const auto& connection : xodr_junction.connections) {
+      participating_road_ids.insert(connection.second.incoming_road);
+      participating_road_ids.insert(connection.second.connecting_road);
+    }
+    for (const std::string& road_id : participating_road_ids) {
```
Comment (jakeaboganda) [inline src/maliput_malidrive/builder/builder_tools.cc:311]: Can we add here a warning if there is a roadTpe conflict?

Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/builder/builder_tools.cc:311]: What do you mean exactly? Multiple road_types in a Junction are allowed, and if one of those were to be a Motorway, then the Junction shan't be an intersection.

Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/builder/builder_tools.cc:311]: Even more, multiple road types in a single road are allowed, since they are delimited by starting `s`: https://publications.pages.asam.net/standards/ASAM_OpenDRIVE/ASAM_OpenDRIVE_Specification/1.8.0/specification/10_roads/10_04_road_type.html

File: test/regression/builder/builder_tools_test.cc (hunk 5 above)
Comment (santiago-lopez-ekumen) [reply test/regression/builder/builder_tools_test.cc:919]: Already there on line 963.

File: src/maliput_malidrive/xodr/parser_configuration.h (hunk 4 above)
Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/xodr/parser_configuration.h:59]: Done.

File: src/maliput_malidrive/builder/builder_tools.h (hunk 3 above)
Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/builder/builder_tools.h]: Done.

File: src/maliput_malidrive/builder/builder_tools.cc (hunk 2 above)
Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/builder/builder_tools.cc:280]: Done.

File: src/maliput_malidrive/builder/builder_tools.cc (hunk 1 above)
Comment (santiago-lopez-ekumen) [reply src/maliput_malidrive/builder/builder_tools.cc]: The comment is meant for the lambda within `DetermineJunctionIntersectionFromXodr`. I updated the comment to be more explicit on what it does.

**Review by JCarosella — APPROVED** [review]

File: src/maliput_malidrive/builder/builder_tools.cc (hunk 6 above)
Comment (jakeaboganda) [reply src/maliput_malidrive/builder/builder_tools.cc:311]: You're absolutely correct -- let me retract the comment. It is possible for multi road_types.
